StyleCop.Analyzers/StyleCop.Analyzers/SpacingRules/SA1027UseTabsCorrectly.cs

```diff
@@ -0,0 +1,268 @@
+
+        private static readonly Action<CompilationStartAnalysisContext> CompilationStartAction = HandleCompilationStart;
+        private static readonly Action<SyntaxTreeAnalysisContext, StyleCopSettings> SyntaxTreeAction = HandleSyntaxTree;
+
+        private static readonly ImmutableDictionary<string, string> ConvertToTabsProperties =
+            ImmutableDictionary.Create<string, string>().SetItem(BehaviorKey, ConvertToTabsBehavior);
+
+        private static readonly ImmutableDictionary<string, string> ConvertToSpacesProperties =
+            ImmutableDictionary.Create<string, string>().SetItem(BehaviorKey, ConvertToSpacesBehavior);
+
+        /// <inheritdoc/>
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics { get; } =
+            ImmutableArray.Create(Descriptor);
+
+        /// <inheritdoc/>
+        public override void Initialize(AnalysisContext context)
+        {
+            context.RegisterCompilationStartAction(CompilationStartAction);
+        }
+
+        private static void HandleCompilationStart(CompilationStartAnalysisContext context)
+        {
+            context.RegisterSyntaxTreeActionHonorExclusions(SyntaxTreeAction);
+        }
+
+        private static void HandleSyntaxTree(SyntaxTreeAnalysisContext context, StyleCopSettings settings)
+        {
+            SyntaxNode root = context.Tree.GetCompilationUnitRoot(context.CancellationToken);
+            ImmutableArray<TextSpan> excludedSpans;
+            if (!LocateExcludedSpans(root, out excludedSpans))
+            {
+                return;
+            }
+
+            ReportIncorrectTabUsage(context, settings.Indentation, excludedSpans);
+        }
+
+        private static void ReportIncorrectTabUsage(SyntaxTreeAnalysisContext context, IndentationSettings indentationSettings, ImmutableArray<TextSpan> excludedSpans)
+        {
+            SyntaxTree syntaxTree = context.Tree;
+            SourceText sourceText = syntaxTree.GetText(context.CancellationToken);
+
+            string completeText = sourceText.ToString();
+            int length = completeText.Length;
+
+            bool useTabs = indentationSettings.UseTabs;
+            int tabSize = indentationSettings.TabSize;
+
+            int excludedSpanIndex = 0;
+            var lastExcludedSpan = new TextSpan(completeText.Length, 0);
+            TextSpan nextExcludedSpan = !excludedSpans.IsEmpty ? excludedSpans[0] : lastExcludedSpan;
+
+            int lastLineStart = 0;
+            for (int startIndex = 0; startIndex < length; startIndex++)
+            {
+                if (startIndex == nextExcludedSpan.Start)
+                {
+                    startIndex = nextExcludedSpan.End - 1;
+                    excludedSpanIndex++;
+                    nextExcludedSpan = excludedSpanIndex < excludedSpans.Length ? excludedSpans[excludedSpanIndex] : lastExcludedSpan;
+                    continue;
```

Comment:
:thought_balloon: The continue statement is unnecessary, as it is impossible to have adjacent excluded spans, because they would have been joined by `ReduceTextSpans`. If `startIndex` is set to `nextExcludedSpan.End`, the flow can just continue.